be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -361,7 +361,8 @@ bool BetaRowsetWriter::_check_and_set_is_doing_segcompaction() {
 Status BetaRowsetWriter::_segcompaction_if_necessary() {
     Status status = Status::OK();
     if (!config::enable_segcompaction || !_context.enable_segcompaction ||
-        !_check_and_set_is_doing_segcompaction()) {
+        !_check_and_set_is_doing_segcompaction() ||
+        !_context.tablet_schema->cluster_key_idxes().empty()) {
```

Comment:
@mymeiyi  This will break seg compaction, causing _segcompacting_cond never notified. I suggest you put your new condition before _check_and_set_is_doing_segcompaction. And do me a favor: add a comment here to warn other developers to leave _check_and_set_is_doing_segcompaction as the last condition.